altair/typing.py

```diff
@@ -46,11 +46,13 @@
     "ChartType",
     "EncodeKwds",
     "Optional",
+    "ThemeConfig",
     "is_chart_type",
 ]
 
 from altair.utils.schemapi import Optional
 from altair.vegalite.v5.api import ChartType, is_chart_type
+from altair.vegalite.v5.schema._config import ThemeConfig
```

Comment:
I'm starting off quite conservative here, since there are many new symbols added in `_config.py`.

You can utilise all the `TypedDict`(s) without actually using the classes, like in https://github.com/vega/altair/pull/3536/files#diff-e52efec94b8b7f0f1b27eac85385173dacfaca37d65afd0df9839e09e58bf377

Personally, I'd find it helpful to have a select few available in `altair.typing` and then the rest from one level down.

So this could look like:

```py

from altair.typing import ThemeConfig, ConfigKwds
from altair.typing.theme import LegendConfigKwds, DateTimeKwds, ...
```

The above is just an example, particularly trying to avoid the conflict with `themes` being a variable

# Related
- https://github.com/vega/altair/issues/2918
- https://github.com/vega/altair/issues/3337

Reply:
Having it in `typing.theme` sounds good to me! I would probably also want to use some of them in my own code, do see the value of having a select few available but totally fine if we tackle this in a follow-up PR, whatever works best for you!

Reply:
> Having it in `typing.theme` sounds good to me! I would probably also want to use some of them in my own code, do see the value of having a select few available but totally fine if we tackle this in a follow-up PR, whatever works best for you!

Perfect!

We can always iron this out before `v5.5.0`, but I built the docs with [`12b6410` (#3536)](https://github.com/vega/altair/pull/3536/commits/12b6410b85d278a051b16cbf3054b6f0a261be83) and it came out fine